st2actions/st2actions/runners/cloudslang/cloudslang_runner.py

```diff
@@ -0,0 +1,109 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+
+import os
+import tempfile
+from oslo.config import cfg
+from eventlet.green import subprocess
+from st2common.util.green.shell import run_command
+from st2common.util.shell import quote_unix
+from st2common import log as logging
+from st2actions.runners import ActionRunner
+from st2common.constants.action import LIVEACTION_STATUS_SUCCEEDED
+from st2common.constants.action import LIVEACTION_STATUS_FAILED
+from st2common.constants.runners import LOCAL_RUNNER_DEFAULT_ACTION_TIMEOUT
+import st2common.util.jsonify as jsonify
+
+
+LOG = logging.getLogger(__name__)
+
+# constants to lookup in runner_parameters.
+RUNNER_PATH = 'flow_path'
+RUNNER_INPUTS = 'inputs'
+RUNNER_TIMEOUT = 'timeout'
+
+
+def get_runner():
+    return CloudSlangRunner(str(uuid.uuid4()))
+
+
+class CloudSlangRunner(ActionRunner):
+    """
+    Runner which executes cloudslang flows and operations as single action
+    """
+    KEYS_TO_TRANSFORM = ['stdout', 'stderr']
+
+    def __init__(self, runner_id):
+        super(CloudSlangRunner, self).__init__(runner_id=runner_id)
+
+    def pre_run(self):
+        self._user = cfg.CONF.system_user.user
+        self._cloudslang_home = cfg.CONF.cloudslang.home_dir
+        self._path = self.runner_parameters.get(RUNNER_PATH)
+        self._inputs = self.runner_parameters.get(RUNNER_INPUTS)
+        self._timeout = self.runner_parameters.get(RUNNER_TIMEOUT,
+                                                   LOCAL_RUNNER_DEFAULT_ACTION_TIMEOUT)
+
+    def run(self, action_parameters):
+        LOG.debug('    action_parameters = %s', action_parameters)
+
+        has_inputs = self._inputs is not None
+        inputs_file = None
+        if has_inputs:
+            inputs_file = tempfile.NamedTemporaryFile()
+            LOG.info(self._inputs)
+            inputs_dict = dict(pair.split("=") for pair in self._inputs.split(","))
+            LOG.info(inputs_dict)
+            import yaml
+            inputs_file.write(yaml.safe_dump(inputs_dict, default_flow_style=False))
```

Comment:
It would be safer to use `with` context manager here because right now we don't close the file and leave it open.

```python
yaml_data = yaml.safe_dump(inputs_dict, default_flow_style=False)
with open(inputs_file, 'w') as fp:
    fp.write(yaml_data)

LOG.info('Inputs file: %s', yaml_data)
```